addons/massCompat/script_component.hpp

```diff
@@ -10,3 +10,220 @@
+MAGAZINE_WELL_RELOAD_TRACER_TYPE(ammoType,bulletCount,Red,magazine), \
+MAGAZINE_WELL_RELOAD_TRACER_TYPE(ammoType,bulletCount,Green,magazine), \
+MAGAZINE_WELL_RELOAD_TRACER_TYPE(ammoType,bulletCount,Yellow,magazine), \
+MAGAZINE_WELL_RELOAD_TRACER_TYPE(ammoType,bulletCount,White,magazine), \
+MAGAZINE_WELL_RELOAD_TRACER_TYPE(ammoType,bulletCount,IR,magazine)
+
+#define MAGAZINE_WELL_BOX(ammoType,bulletCount) \
+MAGAZINE_WELL_TYPE(ammoType,bulletCount,box), \
+MAGAZINE_WELL_TRACER_TYPE(ammoType,bulletCount,Red,box), \
+MAGAZINE_WELL_TRACER_TYPE(ammoType,bulletCount,Green,box), \
+MAGAZINE_WELL_TRACER_TYPE(ammoType,bulletCount,Yellow,box), \
+MAGAZINE_WELL_TRACER_TYPE(ammoType,bulletCount,White,box), \
+MAGAZINE_WELL_TRACER_TYPE(ammoType,bulletCount,IR,box), \
+MAGAZINE_WELL_RELOAD_TRACER_TYPE(ammoType,bulletCount,Red,box), \
+MAGAZINE_WELL_RELOAD_TRACER_TYPE(ammoType,bulletCount,Green,box), \
+MAGAZINE_WELL_RELOAD_TRACER_TYPE(ammoType,bulletCount,Yellow,box), \
+MAGAZINE_WELL_RELOAD_TRACER_TYPE(ammoType,bulletCount,White,box), \
+MAGAZINE_WELL_RELOAD_TRACER_TYPE(ammoType,bulletCount,IR,box)
+
+#define MAGAZINE_WELL_ALT_MAG(ammoType,bulletCount,alt) \
+MAGAZINE_WELL_ALT_TYPE(ammoType,bulletCount,magazine,alt), \
+MAGAZINE_WELL_TRACER_ALT_TYPE(ammoType,bulletCount,Red,magazine,alt), \
+MAGAZINE_WELL_TRACER_ALT_TYPE(ammoType,bulletCount,Green,magazine,alt), \
+MAGAZINE_WELL_TRACER_ALT_TYPE(ammoType,bulletCount,Yellow,magazine,alt), \
+MAGAZINE_WELL_TRACER_ALT_TYPE(ammoType,bulletCount,White,magazine,alt), \
+MAGAZINE_WELL_TRACER_ALT_TYPE(ammoType,bulletCount,IR,magazine,alt), \
+MAGAZINE_WELL_RELOAD_TRACER_ALT_TYPE(ammoType,bulletCount,Red,magazine,alt), \
+MAGAZINE_WELL_RELOAD_TRACER_ALT_TYPE(ammoType,bulletCount,Green,magazine,alt), \
+MAGAZINE_WELL_RELOAD_TRACER_ALT_TYPE(ammoType,bulletCount,Yellow,magazine,alt), \
+MAGAZINE_WELL_RELOAD_TRACER_ALT_TYPE(ammoType,bulletCount,White,magazine,alt), \
+MAGAZINE_WELL_RELOAD_TRACER_ALT_TYPE(ammoType,bulletCount,IR,magazine,alt)
+
+#define MAGAZINE_WELL_ALT_BOX(ammoType,bulletCount,alt) \
+MAGAZINE_WELL_ALT_TYPE(ammoType,bulletCount,box,alt), \
+MAGAZINE_WELL_TRACER_ALT_TYPE(ammoType,bulletCount,Red,box,alt), \
+MAGAZINE_WELL_TRACER_ALT_TYPE(ammoType,bulletCount,Green,box,alt), \
+MAGAZINE_WELL_TRACER_ALT_TYPE(ammoType,bulletCount,Yellow,box,alt), \
+MAGAZINE_WELL_TRACER_ALT_TYPE(ammoType,bulletCount,White,box,alt), \
+MAGAZINE_WELL_TRACER_ALT_TYPE(ammoType,bulletCount,IR,box,alt), \
+MAGAZINE_WELL_RELOAD_TRACER_ALT_TYPE(ammoType,bulletCount,Red,box,alt), \
+MAGAZINE_WELL_RELOAD_TRACER_ALT_TYPE(ammoType,bulletCount,Green,box,alt), \
+MAGAZINE_WELL_RELOAD_TRACER_ALT_TYPE(ammoType,bulletCount,Yellow,box,alt), \
+MAGAZINE_WELL_RELOAD_TRACER_ALT_TYPE(ammoType,bulletCount,White,box,alt), \
+MAGAZINE_WELL_RELOAD_TRACER_ALT_TYPE(ammoType,bulletCount,IR,box,alt)
+
+#define TRACER_CLASSES(baseClass)\
+class AMMO_CLASS(SCORE_3(baseClass,tracer,Red)): AMMO_CLASS(baseClass) { \
```

Comment:
We could add `displaynameshort="Tracer";` to these so it shows in the weapon display box, I think many vanilla mags do this.

Reply:
Is it the bullet or magazine that displays it?

Reply:
must be on the mag, because the ammo isn't changed